### ISSUE TITLE
Incorrect capsule name displayed when ~ is in query string

### ISSUE THREAD

**Issue by acidus99** [skyjake/lagrange#603]
Usually, Lagrange displays the domain name of a capsule at top of the page:

<img width="357" alt="image" src="https://github.com/skyjake/lagrange/assets/4118560/15ad39b2-590b-43e8-aeaf-d50055b92c32">

It appears that there is some special logic to handle `~`, I assume to provide a more accurate capsule name for shared pubnix's:

<img width="448" alt="image" src="https://github.com/skyjake/lagrange/assets/4118560/9f224845-87bd-40f6-af29-8068e615f34f">

However, this logic will display anything after a `~` as the capsule name. There is no logic to just look in the path component of the URL. This means that if a query string contains a `~`, Lagrange displays the contents after it as the capsule name:

<img width="929" alt="image" src="https://github.com/skyjake/lagrange/assets/4118560/6918dea1-7b53-4694-95ee-732d775d4361">

Here is an example URL:
```
gemini://kennedy.gemi.dev/archive/cached?url=gemini%3a%2f%2fwarmedal.se%2f~antenna%2f&t=638181102270000000&raw=False
```

I attempted to work around this by explicitly URL encoding the `~` character in the query string as `%7E`, but that does not resolve the issue.

**Comment by skyjake** [comment]
The intended behavior is to only look at the path component when finding a possible user name.

**Comment by skyjake** [comment]
Fixed for v1.17.
